### PR TITLE
Added BlobDownloadDetails.CreatedOn

### PR DESCRIPTION
**Pull request by seanmcc-msft**
x-ms-creation-time has been returned since x-ms-version 2017-11-09, we missed it originally.

See https://github.com/Azure/azure-rest-api-specs/pull/22559 for more info.

### CODE REVIEW

**Comment by amnguye** [comment]
Resolves https://github.com/Azure/azure-sdk-for-net/issues/33899

**Comment by amnguye** [comment]
@seanmcc-msft 

Could we also make sure that this is in DataLake as well? Looks like we just have to update the BlobDownloadInfo => FileDownloadInfo

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeExtensions.cs#L72

File: sdk/storage/Azure.Storage.Blobs/src/autorest.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/080b332b7572514a2e100dd2fa1fb86cb8edcb08/specification/storage/data-plane/Microsoft.BlobStorage/preview/2021-12-02/blob.json
+    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/83ea737da20c6819531853641a432d01acbe1897/specification/storage/data-plane/Microsoft.BlobStorage/preview/2021-12-02/blob.json
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/src/autorest.md:7]: TODO update this when https://github.com/Azure/azure-rest-api-specs/pull/22559 merges

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/autorest.md (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/autorest.md]: Fixed.

**Comment by amnguye** [comment]
@seanmcc-msft  could you also update the changelog bot both blobs and datalake?

**Pull request merged**